### ISSUE TITLE
Backwards Pass w/Output Gradients

### ISSUE THREAD

**Issue by HapeMask** [facebookarchive/caffe2#1955]
Can caffe2 run a backward pass with externally provided output gradients? I have a loss function that can't be defined in caffe2, but I have the gradients of that loss w.r.t. the output of a net I can define, and want to compute the gradient of the loss w.r.t. the network's input.

As a concrete example, in pytorch I would do something like

```python
gz = th.autograd.grad(net_out, net_in, external_grad)
```

I looked through the API and tutorials but couldn't seem to find a gradient op that would allow this.

**Comment by lironmo** [comment]
+1
